### PR TITLE
Add alibuild and alidist versions to report-pr-errors log

### PR DESCRIPTION
**Pull request by singiamtel**
Should make it easier to debug issues in the pipeline.

It fails gracefully if either command fails, doesn't interrupt the report

### CODE REVIEW

**Comment by singiamtel** [comment]
Example output 

image

**Comment by singiamtel** [comment]
New format (added a link)

image

File: report-pr-errors [hunk 1]
```diff
@@ -21,7 +21,7 @@ from alibot_helpers.github_utilities import (
     calculateMessageHash, setGithubStatus, parseGithubRef, GithubCachedClient,
     PickledCache,
 )
-from alibot_helpers.utilities import to_unicode
+from alibot_helpers.utilities import to_unicode, getout
```
Comment (ktf) [inline report-pr-errors:24]: maybe a better approach would be to expose the function in alibuild to retrieve the version?

File: report-pr-errors [hunk 2]
```diff
@@ -314,6 +317,27 @@ class Logs(object):
...
+            (alibuild_version, status) = getout("alibuild version")
+            if status:
+                self.alibuild_version = "Could not get alibuild version"
+            else:
+                self.alibuild_version = alibuild_version
+        except Exception as e:
+            print("Error while getting alibuild version:", e, file=sys.stderr)
+            self.alibuild_version = "Could not get alibuild version"
+        try:
+            (alidist_version, status) = getout("git -C ./alidist rev-parse --short HEAD")
```
Comment (ktf) [inline report-pr-errors:331]: Do not assume alidist is using git.

**Review by ktf — CHANGES_REQUESTED** [review]
Thank you, I was thinking about something similar actually. I am not sure of the  approach though. Could you check the comments?

File: alibot_helpers/utilities.py [hunk 3]
```diff
@@ -23,3 +25,11 @@ def to_unicode(s):
     elif not isinstance(s, unicode):
         return unicode(str(s))
     return s
+
+
+def getout(cmd):
```
[the remark this answers is not in the document]
Comment (singiamtel) [reply alibot_helpers/utilities.py]: Using subprocess.run instead

File: report-pr-errors (hunk 1 above)
Comment (singiamtel) [reply report-pr-errors]: I'm not sure I get what you mean. Currently in ali-bot we only pickup whatever alibuild we find in $PATH, you'd like to import it as a python module instead?

File: report-pr-errors (hunk 2 above)
Comment (singiamtel) [reply report-pr-errors]: It now fails when alidist is not a git repo. I'm not sure how to get the alidist version when it doesn't have history though

```bash
$ git --git-dir ./alidist/.git rev-parse --short HEAD
f0f86990
$ mv ./alidist/.git ./alidist/.git2
$ git --git-dir ./alidist/.git rev-parse --short HEAD
fatal: not a git repository: './alidist/.git'
```

﻿Screenshot 2024-06-26 at 14 33 47 image

Comment (ktf) [reply report-pr-errors]: Why not simply passing the alibuild version (obtained via `aliBuild version`) and the alidist one as parameters / environment variables? This way you do not need to know how to get them from the script reporting about the builds. You could actually reuse ALIBUILD_SLUG and ALIDIST_SLUG which I think we already define.

Comment (singiamtel) [reply report-pr-errors]: Are they also set for CI jobs? I've only found the definition for the Jenkins jobs.

(if they aren't we can also add them before merging here and just use those)

Comment (singiamtel) [reply report-pr-errors]: Implemented getting the version from the env vars, as discussed

File: alibot_helpers/github_utilities.py [hunk 4]
```diff
@@ -407,3 +407,8 @@ def setGithubStatus(cgh, args, debug_print=True):
         repo_name=repo_name,
         ref=commit_ref
     )
+
+
+def generate_gh_htmllink_from_commit(commit: str, repo: str) -> str:
```
Comment (ktf) [inline alibot_helpers/github_utilities.py:412]: If it's not used elsewhere, I would prefer we define it only where it's needed.

File: report-pr-errors [hunk 5]
```diff
@@ -314,6 +317,22 @@ class Logs(object):
...
+        # The versions are defined differently for both job runner setups
+        # Jenkins (daily builds):
+        #   - aliBuild: ALIBUILD_SLUG
+        #   - alidist: ALIDIST_SLUG (if not set it defaults to alidist@master)
+        # Nomad (CI checks):
+        #   - aliBuild: INSTALL_ALIBUILD
+        #   - alidist: (defaults to alidist@master)
+        self.alibuild_version = os.getenv('ALIBUILD_SLUG') or os.getenv('INSTALL_ALIBUILD') or 'Not pinned (most likely alibuild@master)'
+        alidist_version = os.getenv('ALIDIST_SLUG')
+        if alidist_version:
```
Comment (ktf) [inline report-pr-errors:330]: I am not sure I understand the logic here. Why do you do two different things depending on if it is alibuild_version or alidist version?

Comment (singiamtel) [reply report-pr-errors:335]: This is an old change from when I was getting the commit hash, which was easy to convert to a github URL. The new version is not, so I'll remove the github links

**Comment by singiamtel** [comment]
Should be fixed now

**Review by ktf — APPROVED** [review]

**Pull request merged**